test/new-e2e/tests/windows/iis.go

```diff
@@ -0,0 +1,85 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package windows contains helpers for Windows E2E tests
+package windows
+
+import (
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
+)
+
+// IISSiteDefinition represents an IIS site definition
+type IISSiteDefinition struct {
+	Name        string //  name of the site
+	BindingPort string // port to bind to, of the form '*:8081'
+	AssetsDir   string // directory to copy for assets
+}
+
+// InstallIIS installs IIS on the target machine
+func InstallIIS(host *components.RemoteHost) error {
+	script := `
+	function ExitWithCode($exitcode) {
+		$host.SetShouldExit($exitcode)
+		exit $exitcode
+	  }
+	$result = install-windowsfeature -name Web-Server -IncludeManagementTools
+	if (! $result.Success ) {
+		exit -1
+	}
+	if ($result.RestartNeeded -eq "Yes") {
+		ExitWithCode(3010)
+	}
+	`
+	scriptFile := `c:\temp\install-iis.ps1`
+	err := host.MkdirAll("C:\\temp")
+	if err != nil {
+		return err
+	}
+	_, err = host.WriteFile(scriptFile, []byte(script))
+	if err != nil {
+		return err
+	}
+	// still need to figure out if we need to reboot
+	output, err := host.Execute(scriptFile)
+	if err != nil {
+		return fmt.Errorf("failed to install IIS: %w\n%s", err, output)
+	}
+	return nil
+}
+
+// CreateIISSite creates an IIS site on the target machine
+func CreateIISSite(host *components.RemoteHost, site []IISSiteDefinition) error {
+
+	for _, s := range site {
+
+		// create the site directory
+		//tgtpath := fmt.Sprintf("c:\\tmp\\inetpub\\%s", s.Name)
+		tgtpath := path.Join("c:", "tmp", "inetpub", s.Name)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
This should use [strings.Join](https://pkg.go.dev/strings#Join) with `sep="\\"`

Reply:
as discussed extensively on slack, I don't think it should.
the `.MkdirAll` is expecting the path separator of the test controller host, not the target host.

Reply:
I guess we should change `MkdirAll` to support cross-platform usage cc @clarkb7

Reply:
https://github.com/DataDog/datadog-agent/pull/23209

Reply:
If this is a blocking comment, please provide an example for how you'd like for this to look. If `.MkdirAll` expects the path separator of the test controller, then the path seps should stay as-is.

Happy to abide by best practice - it appears as though we intentionally wanted these to be forward slashes.

@clarkb7, do you know what the path forward should be?

Reply:
> If .MkdirAll expects the path separator of the test controller

See the linked PR for details, but unfortunately the go lib always expects forward slashes, the PR handles this on our side now. So IMO the tests should use whatever the test owner thinks is more readable.